pkg/util/kubernetes/apiserver/services_test.go

```diff
@@ -15,52 +15,83 @@ import (
 	"github.com/stretchr/testify/assert"
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
-type podTest struct {
+type podTestDef struct {
+	uid       string
 	ip        string
 	name      string
 	namespace string
 }
 
-type serviceTest struct {
-	svcName string
-	podIps  []string
+type endPointTestDef struct {
+	name    string
+	subsets [][]addressTestDef
 }
 
-func createSvcList(nodeName string, svcs []serviceTest) v1.EndpointsList {
-	var list v1.EndpointsList
+type addressTestDef struct {
+	ip          string
+	nodeName    string
+	targetPodId string
+}
+
+func mapPodTestDef(defs []podTestDef) map[string]podTestDef {
+	mapped := make(map[string]podTestDef)
+	for _, d := range defs {
+		mapped[d.uid] = d
+	}
+	return mapped
+}
 
-	for _, svc := range svcs {
-		endpoints := v1.Endpoints{}
-		endpointsSubset := v1.EndpointSubset{}
-		ep := v1.EndpointAddress{}
-		endpoints.Name = svc.svcName
+func createEndpointList(nodeName string, defs []endPointTestDef, pods []podTestDef, legacyKube bool) v1.EndpointsList {
```

Comment:
Can we change the name of `legacyKube` to something more descriptive, or at least add a comment for it? Here it looks like 1.3 is considered legacy, that definition may (will) change in the future.